user_guide_src/source/changelogs/v4.3.0.rst

```diff
@@ -12,7 +12,16 @@ Release Date: Unreleased
 Highlights
 **********
 
-- TBD
+- Query Builder supports **upsert()**, **upsertBatch()**, **deleteBatch()** and
+  now ***batch()** methods can set data from a query
+  (*contributed by* `sclubricants <https://github.com/sclubricants>`_).
+  See `Query Builder`_.
+- Database Forge supports to **add indexes in the existing tables** and
+  **name indexes** (*contributed by* `sclubricants <https://github.com/sclubricants>`_).
+  See `Forge`_.
+- To make the default configuration more secure, the default Validation Rules
+  have been changed to **Strict Rules**.
+- The behavior when a database error occurs has been changed. See :ref:`exceptions-when-database-errors-occur`.
 
 BREAKING
 ********
```

Comment:
>- **Model:** ``Model::update()`` method now raises a ``DatabaseException`` if it **generates an SQL
  statement** without a WHERE clause; Model does not support operations that update all records.

Line 99

Should  be **generates a SQL statement**

Reply:
Line 181 and 184 seem to have an unneeded empty line.

Line 183

>- In addition to the added ``$asQuery`` parameter above the following methods also now **return an array**.

Should be:

- In addition to the added ``$asQuery`` parameter above the following methods also now **returns an array**.

Reply:
It seems both `a SQL` and `an SQL` can be used.
https://english.stackexchange.com/questions/5265/an-sql-server-database-schema-or-a-sql-server-database-schema

Reply:
I use `an SQL` when referring to the language because it is an acronym of Structured Query Language and should be pronounced as S-Q-L. If referring to the driver Microsoft has, then it is `a SQL` because it's pronounced as sequel.

`return` is correct because it refers to the `following methods`.

Reply:
> SQL does not begin with a vowel sound.. even if S-Q-L is the way you're saying it.

What do you mean?
I assume to pronounce this way.
https://translate.google.co.jp/?sl=en&tl=ja&text=an%20S%20Q%20L&op=translate

Reply:
> the also suggests plurality so your talking about more than one thing returning something.

I am not a native English speaker and I honestly don't know the difference.
I hope that native speakers can discuss and decide.

Reply:
> SQL does not begin with a vowel sound.. even if S-Q-L is the way you're saying it.

No offense but English is my second language since we're taught with that since grade school. And the [IPA](https://en.wikipedia.org/wiki/International_Phonetic_Alphabet) pronunciation of SQL is either `[/ˈsiːkwəl/]` "sequel" or `[/ˌɛsˌkjuːˈɛl/]` "S-Q-L". If you would listen to the [pronunciation](https://upload.wikimedia.org/wikipedia/commons/5/5f/En-us-SQL.ogg) of the second, it is clearly like "es-q-el" so it starts with a vowel sound. Maybe you're confusing vowel sound with vowel letter? Continuing with your example, "university" starts with the vowel letter "u" but starts with the consonant sound "y", so it is "a university" not "an university".

> the also suggests plurality so your talking about more than one thing returning something.

That's true, and "following methods" is a plural subject so we use the plural form (which is also the base form) of the verb  "return" which is also "return". "returns" is the singular form, isn't it?

Reply:
You could say "an S-Q-L statement" 

This is correct. 

"An S-Q-L" is ambiguous

Reply:
Just out of curiosity for you guys is it:

A SQLite database

Or

An SQLite database

Reply:
It seems to me that English native speakers pronounce "sequel".

In my country, SQL is a foreign word, and it is pronounced like "ess-queue-el" 100%.
It may surprise you but Microsoft SQL Server is pronounced like "ess-queue-el server".

SQLite is like "ess-queue-light".

Reply:
I move between both of them so I'm fine with either solution. I don't think there's a right or wrong here.

Reply:
>The official way to pronounce “MySQL” is “My Ess Que Ell” (not “my sequel”), but we do not mind if you pronounce it as “my sequel” or in some other localized way.

https://dev.mysql.com/doc/refman/8.0/en/what-is-mysql.html#idm139761506951328

Reply:
SQLite creator says pronounce it however you like.

https://english.stackexchange.com/questions/431329/what-is-the-correct-pronunciation-of-sqlite